### ISSUE TITLE
Flashing screen on macOS with latest Ebiten

### ISSUE THREAD

**Issue by lucymhdavies** [lucymhdavies/egg-game#29]
Unsure whether this is me using the API wrong, or whether this is a bug in Ebiten.

The most recent commit of Ebiten which works for me is https://github.com/hajimehoshi/ebiten/commit/a064955a

As of https://github.com/hajimehoshi/ebiten/commit/72be1bde, the screen flashes. It appears that it's swapping between the very first frame and the most recent frame.

I've not found an example of this bug in the ebiten examples, but in egg-game I see the following:

https://youtu.be/ackwd6Z03TA

Annoyingly, I've not seen the issue in any other code of mine, nor have I seen it in any Ebiten examples, which makes it likely that the error is with my code, rather than necessarily being a bug in Ebiten.

For now, I'll pin to https://github.com/hajimehoshi/ebiten/commit/a064955a, and see if I can figure out if this is a problem with my code later.

I've recently updated to Go 1.12, but looks like the issue presents itself in Go 1.11 too.

**Comment by lucymhdavies** [comment]
By way of figuring this one out, do a git bisect on egg-game and see if the issue starts happening at some point.

I've not checked, but looking at the first commit, it'd be very surprising if I could replicate the issue:
https://github.com/lucymhdavies/egg-game/commit/e4804338b79787a97e6262355a91430cb858ac72

**Comment by lucymhdavies** [comment]
With ebiten@master

Egg game e4804338b79787a97e6262355a91430cb858ac72 I cannot replicate the issue.

Time for a git bisect!

```
✔ lucy@fuyuko ~/go/src/github.com/lucymhdavies/egg-game [master|✔]
00:36 $ git bisect start
✔ lucy@fuyuko ~/go/src/github.com/lucymhdavies/egg-game [master|BISECTING|✔]
00:36 $ git bisect bad
✔ lucy@fuyuko ~/go/src/github.com/lucymhdavies/egg-game [master|BISECTING|✔]
00:36 $ git bisect good e4804338b79787a97e6262355a91430cb858ac72
Bisecting: 17 revisions left to test after this (roughly 4 steps)
[6499df45eeb4d997cf0df9bf92b1b6513bd3133b] Initial button: respawning when an egg is dead
✔ lucy@fuyuko ~/go/src/github.com/lucymhdavies/egg-game [:6499df4|BISECTING|✔]
00:37 $ go run .
DEBU[0000] World size: 256, 256
✔ lucy@fuyuko ~/go/src/github.com/lucymhdavies/egg-game [:6499df4|BISECTING|✔]
00:37 $ git bisect bad
Bisecting: 8 revisions left to test after this (roughly 3 steps)
[b4fc6b195d48fae4024d14c24ab123ff4a77a0f9] THE EYES! THEY PIERCE MY SOUL!
✔ lucy@fuyuko ~/go/src/github.com/lucymhdavies/egg-game [:b4fc6b1|BISECTING|✔]
00:37 $ go run .
✔ lucy@fuyuko ~/go/src/github.com/lucymhdavies/egg-game [:b4fc6b1|BISECTING|✔]
00:37 $ git bisect good
Bisecting: 4 revisions left to test after this (roughly 2 steps)
[755635d67fe6eb7fa3c1b0eb778e784b45c9c3eb] Auto-size game based on available screen real estate, but only in JS
✔ lucy@fuyuko ~/go/src/github.com/lucymhdavies/egg-game [:755635d|BISECTING|✔]
00:37 $ go run .
DEBU[0000] World size: 256, 256
✔ lucy@fuyuko ~/go/src/github.com/lucymhdavies/egg-game [:755635d|BISECTING|✔]
00:37 $ git bisect good
Bisecting: 2 revisions left to test after this (roughly 1 step)
[4ad74014504b3d34caf269b7a25c039e649e21c9] Update README.md
✔ lucy@fuyuko ~/go/src/github.com/lucymhdavies/egg-game [:4ad7401|BISECTING|✔]
00:37 $ go run .
DEBU[0000] World size: 256, 256
✔ lucy@fuyuko ~/go/src/github.com/lucymhdavies/egg-game [:4ad7401|BISECTING|✔]
00:37 $ git bisect bad
Bisecting: 0 revisions left to test after this (roughly 0 steps)
[c66526604442b464a172f587464b695dcbb988e8] Draw egg body + eyes offscreen
✔ lucy@fuyuko ~/go/src/github.com/lucymhdavies/egg-game [:c665266|BISECTING|✔]
00:37 $ go run .
DEBU[0000] World size: 256, 256
✔ lucy@fuyuko ~/go/src/github.com/lucymhdavies/egg-game [:c665266|BISECTING|✔]
00:37 $ git bisect bad
c66526604442b464a172f587464b695dcbb988e8 is the first bad commit
commit c66526604442b464a172f587464b695dcbb988e8
Author: Lucy Davinhart <lucy.davies@skybettingandgaming.com>
Date:   Mon Feb 25 22:19:05 2019 +0000

    Draw egg body + eyes offscreen

:040000 040000 8c75285feca08f8ba6a8b9a6997923d72a78b5b5 0860e9ff461a2d9adafa419c87320c9a1ebb8147 M	game
```

c66526604442b464a172f587464b695dcbb988e8 is the first bad commit

**Comment by lucymhdavies** [comment]
And yes, reverting that commit does _technically_ result in the problem going away...

But it re-introduces the eye wobble, which is the reason I added that in the first place.

**Comment by lucymhdavies** [comment]
https://github.com/lucymhdavies/egg-game/tree/GH-29/fix-flashing-screen reverts c66526604442b464a172f587464b695dcbb988e8... but I'm sure there's a better fix.

For example, when drawing the health bar in https://github.com/lucymhdavies/egg-game/commit/f23b247e326ffefc1266e3db149da9f344412fe3, I'm drawing an image off-screen and then drawing that to the screen. That seems to work fine, so there's something I'm doing different here...

**Comment by lucymhdavies** [comment]
This is the problem:

```
	// Create a temporary image to draw body + body parts
	bodyImg = ebiten.NewImageFromImage(egg.images.body, ebiten.FilterDefault)
```
